radixdlt/src/main/java/com/radixdlt/consensus/safety/SafetyViolationException.java

```diff
@@ -0,0 +1,44 @@
+/*
+ *  (C) Copyright 2020 Radix DLT Ltd
+ *
+ *  Radix DLT Ltd licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License.  You may obtain a copy of the
+ *  License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ *  either express or implied.  See the License for the specific
+ *  language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.consensus.safety;
+
+import com.radixdlt.consensus.Vertex;
+
+import java.util.Objects;
+
+/**
+ * A safety violation.
+ */
+public class SafetyViolationException extends Exception {
+	private final Vertex offendingVertex;
+	private final SafetyState safetyState;
+
+	SafetyViolationException(Vertex offendingVertex, SafetyState safetyState, String reason) {
+		super(String.format("vertex %s violates safety at state %s: %s", offendingVertex, safetyState, reason));
+		this.offendingVertex = Objects.requireNonNull(offendingVertex);
+		this.safetyState = new SafetyState(safetyState);
+	}
+
+	public Vertex getOffendingVertex() {

```

Comment:
just `offendingVertex()` ? I think you told me to drop the `get` prefix for getters? :)

Reply:
Yeah, not crucial, but also from the basic-safety-rules branch, so better there.